plugins/woocommerce/src/Caches/OrderCountCache.php

```diff
@@ -23,28 +23,60 @@ class OrderCountCache {
 	protected $expiration = DAY_IN_SECONDS;
 
 	/**
-	 * Cache prefix.
+	 * Retrieves the list of known statuses by order type. A cached array of statuses is saved per order type for
+	 * improved backward compatibility with some of the extensions that don't register all statuses they use with
+	 * WooCommerce.
 	 *
-	 * @var string
+	 * @param string $order_type The type of order.
+	 *
+	 * @return string[]
 	 */
-	private $cache_prefix = 'order-count';
+	private function get_saved_statuses_for_type( string $order_type ) {
+		$statuses = wp_cache_get( $this->get_saved_statuses_cache_key( $order_type ) );
+		if ( ! is_array( $statuses ) ) {
+			$statuses = array();
+		}
+
+		return $statuses;
+	}
 
 	/**
-	 * Get valid statuses.
+	 * Adds the given statuses to the cached statuses array for the order type if they are not already stored.
 	 *
-	 * @return string[]
+	 * @param string   $order_type     The order type to save with.
+	 * @param string[] $order_statuses One or more normalised statuses to add.
+	 *
+	 * @return void
 	 */
-	private function get_valid_statuses() {
-		return array_merge(
-			array_keys( wc_get_order_statuses() ),
-			array_keys( get_post_stati() ),
-		);
+	private function ensure_statuses_for_type( string $order_type, array $order_statuses ) {
+		if ( empty( $order_statuses ) ) {
+			return;
+		}
+
+		$existing     = $this->get_saved_statuses_for_type( $order_type );
+		$new_statuses = array_diff( $order_statuses, $existing );
+		if ( empty( $new_statuses ) ) {
+			return;
+		}
+		$merged = array_unique( array_merge( $existing, $new_statuses ) );
+
+		wp_cache_set( $this->get_saved_statuses_cache_key( $order_type ), $merged, '', $this->expiration );
 	}
 
+
+	/**
+	 * Cache prefix.
+	 *
+	 * @var string
+	 */
+	private $cache_prefix = 'order-count';
+
```

Comment:
Any reason to "move" the property declaration here? Not a problem per se, but a bit non-standard?